app/models/course/material/folder.rb

```diff
@@ -17,8 +17,10 @@ def files_attributes=(files)
     end
   end
 
-  def self.after_course_create(course)
-    course.material_folders.create(name: 'Root')
+  def self.before_course_validation(course)
+    return if course.persisted?
+
+    course.material_folders.build(name: 'Root') if course.material_folders.empty?
```

Comment:
`if` is because the validation runs multiple times. this is for preventing that.

But as you suggested, if use `after_initialise` this can be removed.